processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -157,6 +163,16 @@ protected void initAggs(
               concurrentEventAdd
           )
       );
+      if (isPreserveExistingMetrics()) {
```

Comment:
nit: since this is happening basically per column per row, `isPreserveExistingMetrics()` should probably be saved in a final field or just made `protected` in `IncrementalIndex`, and use that directly which probably has a better chance of being optimized by jvm than this being a method call everywhere (though maybe jvm is cool enough it never usually hurts to help it out by being explicit and cutting down method invocations). same applies to other uses of `isPreserveExistingMetrics()`

side note: it might be worth sanity check measuring performance of ingestion before and after this change to make sure there is little/no impact when this mode is on

Reply:
Done